src/core/libmaven/csvreports.h

```diff
@@ -22,122 +20,65 @@ class CSVReports {
     *such as rt,mz, samples used, compound it, formula, etc
     */
 public:
-    /**
-    *empty constructor
-    */
+    enum ExportType{ GroupExport, PeakExport};
     CSVReports();
+
     /**
-    *@brief-    creating CSVReports by all the samples uploaded
-    *@see -     see details in its definition
-    */
-    CSVReports(vector<mzSample*>& insamples);
+     * CSV exporting is a four step process-
+     * 1. Create one instance of this class by giving required value of
+     * all parameters in below constructor.      
+     * 2. Add all groups one by one to be exported by calling addItem method of this class
+     * 3. call exportGroup() method of this class which will return true if report is successful
+     *  otherwise false
+     * 4. If returned value from exportGroup() is false call getErrorReport method to get error message
+     */ 
     /**
-    *@brief-    destructor, just close all open output files opened for writing csv or tab file
-    */
+     * Note: don't use same instance of this class to export more that one csv file.
```

Comment:
To me it does not make sense to have multiple instances of CsvReports. 
Only one instance of CsvReports should be sufficient.

Reply:
@rish9511 Does it not make sense to have multiple instances to deal with potential multithreading? E.g. the user could initiate two exports at the same time.

Reply:
@chubukov  having multiple instances is not always the way to get away from multi threading issues. We can use locks 
and IIRC export functionality is not multi threaded

Reply:
@rish ok sure

Reply:
@chubukov you tagged the wrong person :grin:

Reply:
@rish9511 @chubukov  We can use same instance, if we can add a reset function to reset all parameters or required parameters including variable holding groups. But this will introduce chances of erroneous result in csv report and overhead on writing code for using that method. Currently, we don't have many requirement to use this class, that's why I didn't write that function. If need be, we'll write that one, however I'll not recommend to use same instance to export multiple csv file to avoid error in result.